protocols/raft/src/main/java/io/atomix/protocols/raft/storage/snapshot/Snapshot.java

```diff
@@ -78,6 +80,17 @@
   /** Deletes the snapshot. */
   void delete();
 
+  /** returns a path to the snapshot location */
+  Path getPath();
+
+  @Override
+  default int compareTo(Snapshot other) {
+    return Comparator.comparingLong(Snapshot::index)
+        .thenComparingLong(Snapshot::term)
+        .thenComparing(Snapshot::timestamp)
+        .compare(this, other);
```

Comment:
I was asking my self whether this could end in a endless loop ? :sweat_smile:

Reply:
Let's discuss it in the office